HLTrigger/Egamma/plugins/HLTPMMassFilter.cc

```diff
@@ -134,38 +129,37 @@ bool HLTPMMassFilter::hltFilter(edm::Event& iEvent,
 
       TLorentzVector pscEle = approxMomAtVtx(theMagField, vertexPos, sc, -1);
       pEleCh2.push_back(pscEle);
-      etaOrig.push_back(sc->eta());
     }
 
+    std::vector<bool> save_cand(scs.size(), true);
     for (unsigned int jj = 0; jj < scs.size(); jj++) {
-      TLorentzVector p1Ele = pEleCh1.at(jj);
-      for (unsigned int ii = 0; ii < scs.size(); ii++) {
-        TLorentzVector p2Ele = pEleCh2.at(ii);
-
-        if (fabs(p1Ele.E() - p2Ele.E()) < 0.00001)
-          continue;
-
-        TLorentzVector pTot = p1Ele + p2Ele;
-        double mass = pTot.M();
-
-        if (mass >= lowerMassCut_ && mass <= upperMassCut_) {
+      for (unsigned int ii = jj + 1; ii < scs.size(); ii++) {
+        if (isGoodPair(pEleCh1[jj], pEleCh2[ii]) or isGoodPair(pEleCh2[jj], pEleCh1[ii])) {
```

Comment:
`if (isGoodPair(pEleCh1[jj], pEleCh2[ii]) ) {` will suffice, and the `or` with `isGoodPair(pEleCh2[jj], pEleCh1[ii])` can be dropped; because the way `isGoodPair` function is written, it does not differentiate between electron and positron

Reply:
or maybe keep it to be safe, running on 1K Zee events I do not see any change in results with and w/o the `or`, but I'm not sure if low-pT would be affected by a wrong charge hypothesis. The way it is now seems safer approach.